File: web/includes/MSTT_Monitor/classes/check.php
<?php
    namespace MSTT_MONITOR\Classes;
    include_once('../common.php');

    class Check extends \MSTT_MONITOR\Common {
        private $id = -1; // Mysql ID
        private $clientID = -1; // Client ID
        private $commandID = -1; // Command ID
        private $timestamp = -1; // Timestamp (Date)
        private $checked = false; // Checked (bool)
        private $error = false; // Error (bool)
        private $finished = false; // Finished (bool)
        private $response = array(); // Response (JSON)

        public function __construct($check) {
            $this->id = $check["id"];
            $this->clientID = $check["client_id"];
            $this->commandID = $check["command_id"];
            $this->timestamp = strtotime($check["timestamp"]);
            $this->checked = $check["checked"];
            $this->error = $check["error"];
            $this->finished = $check["finished"];
            $this->response = json_decode($check["response"], true);
        }

        public function getID() {
            return $this->id;
        }

        public function getClientID() {
            return $this->clientID;
        }

        public function getCommandID() {
            return $this->commandID;
        }

        public function getTimestamp() {
            return $this->timestamp;
        }

        public function getChecked() {
            return $this->checked;
        }

        public function getError() {
            return $this->error;
        }

        public function getFinished() {
            return $this->finished;
        }

        public function getResponse() {
            return $this->response;
        }
    }
?>

File: ny server/models/check.go
package models

import (
	"sync"
	"time"
)

type Check struct {
	rw            *sync.RWMutex
	Command       *Command  `db:"-"`
	PastID        int       `db:"id, primarykey, autoincrement"`
	NextTimestamp time.Time `db:"-"`
	Checked       bool      `db:"checked"`
	Err           bool      `db:"err"`
	Finished      bool      `db:"finished"`
}

// GetCommand
func (c *Check) GetCommand() (cmd *Command) {
	c.rw.RLock()
	cmd = c.Command
	c.rw.RUnlock()
	return
}

// GetID
func (c *Check) GetID() (id int) {
	c.rw.RLock()
	id = c.PastID
	c.rw.RUnlock()
	return
}

// GetTimestamp
func (c *Check) GetTimestamp() (timestamp time.Time) {
	c.rw.RLock()
	timestamp = c.NextTimestamp
	c.rw.RUnlock()
	return
}

// GetChecked
func (c *Check) GetChecked() (checked bool) {
	c.rw.RLock()
	checked = c.Checked
	c.rw.RUnlock()
	return
}

// GetError
func (c *Check) GetError() (err bool) {
	c.rw.RLock()
	err = c.Err
	c.rw.RUnlock()
	return
}

// GetFinished
func (c *Check) GetFinished() (finished bool) {
	c.rw.RLock()
	finished = c.Finished
	c.rw.RUnlock()
	return
}
